### PR TITLE
Add configuration for init/clear function expressions of boxed types

### PR DESCRIPTION
**Pull request by sdroege**
Fixes https://github.com/gtk-rs/glib/issues/469

### CODE REVIEW

**Comment by EPashkin** [comment]
Please update Readme.txt maybe with `TreeIter` as example

File: src/codegen/general.rs [hunk 1]
```diff
@@ -271,6 +283,14 @@ pub fn define_auto_boxed_type(
         "\t\tfree => |ptr| gobject_sys::g_boxed_free({}::{}(), ptr as *mut _),",
         sys_crate_name, get_type_fn
     )?;
+
+    if let (Some(ref init_function_expression), Some(ref clear_function_expression)) =
```
Comment (EPashkin) [inline src/codegen/general.rs]: `ref` not needed anymore in this case

**Comment by sdroege** [comment]
> Please update Readme.txt maybe with `TreeIter` as example

Done

File: src/codegen/general.rs (hunk 1 above)
Comment (sdroege) [reply src/codegen/general.rs]: Done

File: src/config/gobjects.rs [hunk 2]
```diff
@@ -308,6 +307,22 @@ fn parse_object(
...
+        .lookup("init_function_expression")
+        .and_then(Value::as_str)
+        .map(ToOwned::to_owned);
+    let clear_function_expression = toml_object
+        .lookup("clear_function_expression")
+        .and_then(Value::as_str)
+        .map(ToOwned::to_owned);
+
+    if (init_function_expression.is_some() && clear_function_expression.is_none())
+        || (init_function_expression.is_none() && clear_function_expression.is_some())
```
Comment (EPashkin) [inline src/config/gobjects.rs]: Maybe `init_function_expression.is_some() != clear_function_expression.is_some()` ?

Comment (sdroege) [reply src/config/gobjects.rs]: I dislike comparing booleans with `==` and `!=` :)

Comment (EPashkin) [reply src/config/gobjects.rs]: Ok, then I wait for CI and merge

**Comment by EPashkin** [comment]
@sdroege Thanks

**Pull request merged**